crates/subspace-service/src/dsn/import_blocks/root_blocks.rs

```diff
@@ -126,10 +126,9 @@ impl RootBlockHandler {
             }
         }
 
-        for handle in peer_handles {
-            if let Err(err) = handle.await {
-                error!(?err, "Task for root blocks returned an error");
-            }
+        // Driving future task forward.
+        for task in peer_block_tasks {
```

Comment:
Yes, but this does it sequentially. What we want is to do that concurrently. You can use ordered or unordered futures collection for that.